pandapower/toolbox.py

```diff
@@ -1801,13 +1802,18 @@ def adapt_element_idx_references(net, element, element_type, offset=0):
                       for ix in net2["line_geodata"].index]
                 net2.line_geodata.set_index(np.array(ni), inplace=True)
             ignore_index = element not in ("bus", "res_bus", "bus_geodata", "line_geodata", "group")
+            ignore_index = ignore_index & (not retain_original_indices_in_net1)
             dtypes = net[element].dtypes
             try:
                 net[element] = pd.concat([net[element], net2[element]], ignore_index=ignore_index,
                                          sort=False)
             except:
                 # pandas legacy < 0.21
                 net[element] = pd.concat([net[element], net2[element]], ignore_index=ignore_index)
+            if element not in ("bus", "res_bus", "bus_geodata", "line_geodata", "group") and not ignore_index:
```

Comment:
What is this if-condition for? It is never applied because the condition will be never fullfilled.